### PR TITLE
Improve FE tests resiliency

### PR DESCRIPTION
**Pull request by acelaya**
This PR is an attempt to reduce the amount of random test failures during CI, for some tests which are asserting on the result of DOM events which have asynchronous callbacks.

The change in both tests implies listening for the same event in the test itself, and waiting for it to be dispatched, before trying to make any assertions.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/test/FilePickerApp-test.js [hunk 1]
```diff
@@ -210,8 +210,14 @@ describe('FilePickerApp', () => {
...
+        form.addEventListener('submit', resolve)
+      );
+
       // Simulate implicit form submission, as if pressing Enter in title field.
-      wrapper.find('form').getDOMNode().requestSubmit();
+      form.requestSubmit();
+      await waitForFormSubmit;
 
       await waitFor(() => fakeAPICall.called);
       await waitFor(() => onSubmit.called);
```
Comment (acelaya) [inline lms/static/scripts/frontend_apps/components/test/FilePickerApp-test.js:223]: I initially thought we would be able to get rid of these `waitFor` calls, but they seem to still be needed. That makes me have some doubts that my changes actually fixed this, but I think it's worth trying.

**Comment by robertknight** [comment]
> This PR is an attempt to reduce the amount of random test failures during CI, for some tests which are asserting on the result of DOM events which have asynchronous callbacks.

Can you link to some of the CI runs that failed please. The changes in this PR add some extra waits in tests, but I don't think they are doing what the code intends, or put another way, there are simpler changes that would be equivalent.

**Comment by acelaya** [comment]
> > This PR is an attempt to reduce the amount of random test failures during CI, for some tests which are asserting on the result of DOM events which have asynchronous callbacks.
> 
> Can you link to some of the CI runs that failed please. The changes in this PR add some extra waits in tests, but I don't think they are doing what the code intends, or put another way, there are simpler changes that would be equivalent.

These are some examples:

* https://github.com/hypothesis/lms/actions/runs/6396634634/job/17362848646
* https://github.com/hypothesis/lms/actions/runs/6402881799/job/17380298514

**Review by robertknight — COMMENTED** [review]
Based on my understanding, the changes here are simply equivalent to setting a longer timeout in the `waitFor` calls. This timeout is currently very short, at only 10ms by default.

I think what we probably want to do is simply increase the default timeout, and maybe have separate thresholds for warning about long waits, which slow the test suite down, and timeouts which actually cause a failure. I can take a look at this.

**Pull request closed without merging**